### PR TITLE
Plumb --credentials-mode-manual into cloud builder.

### PR DESCRIPTION
**Pull request by abutcher**
[HIVE-2030](https://issues.redhat.com//browse/HIVE-2030)

### CODE REVIEW

File: pkg/clusterresource/alibabacloud.go [hunk 1]
```diff
@@ -88,10 +88,6 @@ func (p *AlibabaCloudBuilder) addInstallConfigPlatform(o *Builder, ic *installer
 	}
 	ic.ControlPlane.Platform.AlibabaCloud = mppMaster
 	ic.Compute[0].Platform.AlibabaCloud = mppWorker
-
-	// Alibaba Cloud only supports manual credentials mode. Manifests including required secrets
-	// must be passed to hive via cd.spec.provisioning.manifestsConfigmapRef
-	ic.CredentialsMode = installertypes.ManualCredentialsMode
```
Comment (wking) [inline pkg/clusterresource/alibabacloud.go:94]: we'll want something that tracks "All Alibaba clusters need to be manual", etc., without requiring folks to set the command-line flag for those platforms.

Comment (abutcher) [reply pkg/clusterresource/alibabacloud.go:94]: Pushed an update that defaults manual mode for Alibaba and IBM Cloud when unset along with an `Info` log that we've defaulted it on the caller's behalf.

I don't know why this error is indented so much. :man_shrugging: 

```
./bin/hiveutil create-cluster --cloud=ibmcloud abutcher-ibmcloud
DEBU[0000] Couldn't find install logs provider environment variable. Skipping. 
INFO[0000] Defaulting credentials mode to Manual for cloud=ibmcloud  command=create-cluster
FATA[0000] Error                                         command=create-cluster error="Manual credentials mode requires --manifests containing custom Secrets with manually provisioned credentials"
```

Comment (2uasimojo) [reply pkg/clusterresource/alibabacloud.go:94]: Well, it isn't because you're using Errorf even though there's no format :shrug:

Comment (abutcher) [reply pkg/clusterresource/alibabacloud.go:94]: It's the `WithError(err)` wrapping.

```diff
-                               opt.log.WithError(err).Fatal("Error")
+                               opt.log.Fatal(err)
```

```
./bin/hiveutil create-cluster --cloud=ibmcloud abutcher-ibmcloud                                                    
DEBU[0000] Couldn't find install logs provider environment variable. Skipping. 
INFO[0000] Defaulting credentials mode to Manual for cloud=ibmcloud  command=create-cluster
FATA[0000] Manual credentials mode requires --manifests containing custom Secrets with manually provisioned credentials  command=create-cluster
```

Comment (abutcher) [reply pkg/clusterresource/alibabacloud.go:94]: I'm tempted to remove the `WithError(err)` like above but probably separate to this change. Anyone have opinions about the UX of that log?

Comment (2uasimojo) [reply pkg/clusterresource/alibabacloud.go:94]: Shrug. It's eye-catching :)

File: contrib/pkg/createcluster/create.go [hunk 2]
```diff
@@ -278,8 +290,9 @@ create-cluster CLUSTER_DEPLOYMENT_NAME --cloud=ovirt --ovirt-api-vip 192.168.1.2
 	flags.StringVar(&opt.HibernateAfter, "hibernate-after", "", "Automatically hibernate the cluster whenever it has been running for the given duration")
 	flags.StringVar(&opt.PullSecretFile, "pull-secret-file", defaultPullSecretFile, "Pull secret file for cluster")
 	flags.StringVar(&opt.BoundServiceAccountSigningKeyFile, "bound-service-account-signing-key-file", "", "Private service account signing key (often created with ccoutil create key-pair)")
-	flags.BoolVar(&opt.CredentialsModeManual, "credentials-mode-manual", false, "Configure the Cloud Credential Operator in the target cluster to Manual mode. Implies the use of --manifests to inject custom Secrets for all CredentialsRequests in the cluster.")
-
+	flags.BoolVar(&opt.CredentialsModeManual, "credentials-mode-manual", false, fmt.Sprintf(`Configure the Cloud Credential Operator in the target cluster to Manual mode.
+Implies the use of --manifests to inject custom Secrets for all CredentialsRequests in the cluster.
+Manual credentials mode is enabled by default and is required for the following clouds: %s`, strings.Join(manualModeClouds, "|")))
```
Comment (2uasimojo) [inline contrib/pkg/createcluster/create.go:295]: The wording of this is a bit confusing. Defaulting implies that there's another option; "required" implies you have to specify the argument, which somewhat contradicts the defaulting and also I think isn't true. What about:
```suggestion
This option is redundant (but permitted) for following clouds, which always use manual mode: %s`, strings.Join(manualModeClouds, "|")))
```

File: contrib/pkg/createcluster/create.go [hunk 3]
```diff
@@ -383,6 +396,11 @@ func (o *Options) Complete(cmd *cobra.Command, args []string) error {
 		o.HibernateAfterDur = &dur
 	}
 
+	if manualCloud := manualCCOModeClouds[o.Cloud]; manualCloud && !o.CredentialsModeManual {
+		o.CredentialsModeManual = true
+		o.log.Infof("Defaulting credentials mode to Manual for cloud=%s", o.Cloud)
```
Comment (2uasimojo) [inline contrib/pkg/createcluster/create.go:401]: As above, "Defaulting" is a bit misleading.
```suggestion
		o.log.Infof("Using Manual credentials mode for cloud=%s", o.Cloud)
```

**Review by 2uasimojo — COMMENTED** [review]
couple wording nits

**Pull request merged**